code/modules/unit_tests/status_effect_validity.dm

```diff
@@ -11,19 +11,16 @@
 		if(tick_speed == INFINITY)
 			TEST_FAIL("Status effect [checking] has tick_interval set to INFINITY, this is not how you prevent ticks - use tick_interval = STATUS_EFFECT_NO_TICK instead.")
 			continue
-		if(tick_speed == 0)
-			TEST_FAIL("Status effect [checking] has tick_interval set to 0, this is not how you prevent ticks - use tick_interval = STATUS_EFFECT_NO_TICK instead.")
-			continue
 		switch(initial(checking.processing_speed))
 			if(STATUS_EFFECT_FAST_PROCESS)
-				if(tick_speed < SSfastprocess.wait)
+				if(tick_speed < SSfastprocess.wait && tick_speed != STATUS_EFFECT_AUTO_TICK)
```

Comment:
Why update all these `if`s rather than just doing `if(tick_speed == STATUS_EFFECT_AUTO_TICK) continue` outside the switch?

Reply:
to avoid skipping the invalid processing speed check at the end of the switch statement